css-inline/src/lib.rs

```diff
@@ -459,8 +459,29 @@ impl<'a> CSSInliner<'a> {
                         for (name, value) in &declarations[*start..*end] {
                             match element_styles.entry(name.as_ref()) {
                                 indexmap::map::Entry::Occupied(mut entry) => {
-                                    if entry.get().0 <= specificity {
-                                        entry.insert((specificity, *value));
+                                    match (
+                                        value.strip_suffix("!important"),
```

Comment:
I'm a little uncertain whether this is the best way to go about this — I considered that we might memoize this test and store an extra boolean in the `declarations` item. however, I think it's probably the case that most of the time there is no overlap, and in cases where there is overlap, there are not many overlaps, so it's probably cheaper to just check it on collision every time. What do you think?